src/ray/gcs/gcs_server/gcs_placement_group_scheduler.h

```diff
@@ -45,23 +45,36 @@ class GcsPlacementGroupSchedulerInterface {
   /// Schedule the specified placement_group.
   ///
   /// \param placement_group to be scheduled.
+  /// \param failure_callback This function is called if the schedule is failed.
+  /// \param success_callback This function is called if the schedule is successful.
   virtual void Schedule(
       std::shared_ptr<GcsPlacementGroup> placement_group,
-      std::function<void(std::shared_ptr<GcsPlacementGroup>)> schedule_failure_handler,
-      std::function<void(std::shared_ptr<GcsPlacementGroup>)>
-          schedule_success_handler) = 0;
+      std::function<void(std::shared_ptr<GcsPlacementGroup>)> failure_callback,
+      std::function<void(std::shared_ptr<GcsPlacementGroup>)> success_callback) = 0;
+
+  /// Get bundles belong to the specified node.
+  ///
+  /// \param node_id ID of the dead node.
+  /// \return The bundles belong to the dead node.
+  virtual absl::flat_hash_map<PlacementGroupID, std::vector<int64_t>> GetBundlesOnNode(
+      const ClientID &node_id) = 0;
 
   virtual ~GcsPlacementGroupSchedulerInterface() {}
 };
 
 class ScheduleContext {
  public:
   ScheduleContext(std::shared_ptr<absl::flat_hash_map<ClientID, int64_t>> node_to_bundles,
+                  const absl::flat_hash_map<int64_t, ClientID> &bundle_locations,
                   const GcsNodeManager &node_manager)
-      : node_to_bundles_(std::move(node_to_bundles)), node_manager_(node_manager) {}
+      : node_to_bundles_(std::move(node_to_bundles)),
+        bundle_locations_(bundle_locations),
+        node_manager_(node_manager) {}
 
   // Key is node id, value is the number of bundles on the node.
   std::shared_ptr<absl::flat_hash_map<ClientID, int64_t>> node_to_bundles_;
+  // Distribution of bundles in nodes. Key is bundle index, value is node id.
+  const absl::flat_hash_map<int64_t, ClientID> &bundle_locations_;
```

Comment:
Can you clarify in the comment that this is the location of existing bundles for this placement group? (I assume)